EK_ELECTROSTATIC_COUPLING

The `EK_ELECTROSTATIC_COUPLING` flag was removed in #2490 but accidentally re-introduced later:
* in 35efe8cd0cdd https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/src/core/forces.cpp#L182-L184
* in 0309058c4f https://github.com/espressomd/espresso/blob/25cefff6ea21a5c1b0f0735fa7b32ec40995ff17/maintainer/configs/no_rotation.hpp#L53

Compilation fails with
```sh
cp ../maintainer/configs/no_rotation.hpp myconfig.hpp
cmake .. -DWITH_CUDA=ON
make
```
```
Unknown feature 'EK_ELECTROSTATIC_COUPLING'
There were errors in '/work/jgrad/espresso-benchmark/build6/src/config/myconfig-final.hpp'
src/config/CMakeFiles/check_myconfig.dir/build.make:57: recipe for target 'src/config/CMakeFiles/check_myconfig' failed
make[2]: *** [src/config/CMakeFiles/check_myconfig] Error 1
```

--- a/maintainer/configs/no_rotation.hpp
+++ b/maintainer/configs/no_rotation.hpp
@@ -50,7 +50,6 @@
 #ifdef CUDA
 #define ELECTROKINETICS
 #define EK_BOUNDARIES
-#define EK_ELECTROSTATIC_COUPLING
 #endif
 
 // Force/energy calculation

--- a/src/core/forces.cpp
+++ b/src/core/forces.cpp
@@ -179,7 +179,7 @@ void calc_long_range_forces() {
   Coulomb::calc_long_range_force();
 /* If enabled, calculate electrostatics contribution from electrokinetics
  * species. */
-#ifdef EK_ELECTROSTATIC_COUPLING
+#ifdef ELECTROKINETICS
   ek_calculate_electrostatic_coupling();
 #endif
 

--- a/src/core/forces.cpp
+++ b/src/core/forces.cpp
@@ -177,11 +177,6 @@ void calc_long_range_forces() {
 #ifdef ELECTROSTATICS
   /* calculate k-space part of electrostatic interaction. */
   Coulomb::calc_long_range_force();
-/* If enabled, calculate electrostatics contribution from electrokinetics
- * species. */
-#ifdef ELECTROKINETICS
-  ek_calculate_electrostatic_coupling();
-#endif
 
 #endif /*ifdef ELECTROSTATICS */
 

